motionless.py

```diff
@@ -121,9 +134,20 @@ def _get_sensor(self):
     def _get_key(self):
         if self.key:
             return ''.join(['key=', self.key, '&'])
+        elif self.clientid:
+            return ''.join(['client=', self.clientid, '&'])
         else:
             return ''
 
+    def _sign(self, url):
+        if self.secret:
+            decoded_key = base64.urlsafe_b64decode(self.secret)
+            # (normalise URL before signing - fails Google's signature checks otherwise)
+            signature = hmac.new(decoded_key, url.replace('%7E', '~').encode('utf-8'), hashlib.sha1)
```

Comment:
```python
url.replace('%7E', '~')
```
This is odd, will it work when using proxies such as GMail's GoogleUserContent?

Reply:
I had to handle that specific case because Python's quote function by default incorrectly encodes the `~` char.  I could have added it to the `safe` parameter of every call - but I thought it was better to leave the actual URLs as they are presumably working currently.

This came up specifically because I had a static map URL with a `~` in it, and Google normalises it by decoding that encoding before doing it's own signature for comparison, so I had to do that so the signatures would match and Google would accept the request.  I followed the advice at https://en.wikipedia.org/wiki/URL_normalization re decoding it, as that is what Google seems to be doing during it's signature comparison.